R/tag_query.R

```diff
@@ -985,17 +997,11 @@ wrapWithRootTag <- function(x) {
 # Return a tag env, tagList(tag envs), or NULL
 tagQueryGetRoot <- function(root) {
   children <- root$children
-  len <- length(children)
-  if (len == 1) {
-    children[[1]]
-  } else if (len > 1) {
-    do.call(tagList, children)
-  } else {
-    # no children?
-    NULL
-  }
+  do.call(visibleTagList, children)
```

Comment:
Could this just be `tagList(root$children)`? I'm not in love with the fact that you get the `[[` indexing for `Root`.

Reply:
Also a comment about `root` being a pseudo-tag might be helpful here. Also, perhaps it makes sense to drop the pseudo-tag data structure for a tagList() ?

Reply:
> perhaps it makes sense to drop the pseudo-tag data structure for a tagList()?

We can't easily edit a tagList in place. We can easily edit a tag env in place. A pseudo root is preferred over `<<-`.  

-----------------

Using a pseudo root tag helps with trying to make a new tagQuery object from a set of tagEnvs. For each tagEnv, we can look for the pseudoRoot to know they come from the same original tagQuery object. If all tagEnvs have the same pseudoRoot tag env, then we can create a new tagQuery object with the selected elements.  If there is a mix of root tags, then an error is thrown.  

This check would not be possible without a pseudoRoot tag env.